### PR TITLE
Dont Merge: IoTHub Extension

### PR DESCRIPTION
**Pull request by mamaso**
@MikeStall

### CODE REVIEW

File: src/WebJobs.Extensions.IoTHub/MessageAsyncCollector.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        public MessageAsyncCollector(ServiceClient client, string deviceId)
+        {
+            _client = client;
+            _deviceId = deviceId;
+        }
+
+        public Task AddAsync(Message item, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return _client.SendAsync(_deviceId, item);
```
Comment (MikeStall) [inline src/WebJobs.Extensions.IoTHub/MessageAsyncCollector.cs]: could this SendAsync  take the cancelation token?

File: src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    /// <summary>
+    /// Binding provider for IoTHub
+    /// </summary>
+    public class IoTHubConfigProvider : IExtensionConfigProvider,
+        IConverter<byte[], Message>,
+        IConverter<string, Message>,
+        IConverter<JObject, Message>,
+        IConverter<IoTHubAttribute, ServiceClient>
+    {
+        internal const string IoTHubConnectionStringName = "AzureWebJobsIoTHub";
```
Comment (MikeStall) [inline src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs]: where is this used?

File: src/WebJobs.Extensions.IoTHub/MessageAsyncCollector.cs (hunk 1 above)
Comment (mamaso) [inline src/WebJobs.Extensions.IoTHub/MessageAsyncCollector.cs]: nope, no method support for cancellation tokens

File: src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs (hunk 2 above)
Comment (mamaso) [inline src/WebJobs.Extensions.IoTHub/IoTHubConfigProvider.cs]: line 38, `nameResolver.Resolve(IoTHubConnectionStringName)`, as well as in the `ValidateConnection` error message

**Comment by mathewc** [comment]
You guys might consider using a branch to work on this together, rather than a public PR, which makes this seem like something actionable that needs to be reviewed.

File: src/WebJobs.Extensions.IoTHub/Converters/JObjectToMessage.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            JToken body = null;
+            var message = input.ToObject<Message>();
+
+            // by convention, use a 'body' property to initialize method
+            if (input.TryGetValue("body", StringComparison.OrdinalIgnoreCase, out body))
+            {
+                // can only set body through constructor
+                var messageWithBody = new Message(Encoding.UTF8.GetBytes((string)body));
+                messageWithBody.Ack = message.Ack;
```
Comment (MikeStall) [inline src/WebJobs.Extensions.IoTHub/Converters/JObjectToMessage.cs]: You may be able to use  Populate() to simplify this: 
    JsonConvert.PopulateObject(input.ToString(), msg);

**Comment by mathewc** [comment]
If you guys are using this PR to review code together, please close this PR and use a branch that both you guys can party in outside of our public PR list please :)

**Pull request closed without merging**